### PR TITLE
Refactor to paymaster url & added check for chain ids + tests

### PR DESCRIPTION
**Pull request by VGabriel45**
# Summary

Refactor to paymaster url & added check for chain ids + tests.
This will introduce a **breaking change**, paymasterUrl will be used over biconomyPaymasterApiKey.

## Change Type

- [ ] Bug Fix
- [x] Refactor
- [ ] New Feature
- [x] Breaking Change
- [ ] Documentation Update
- [ ] Performance Improvement
- [ ] Other

# Checklist

- [x] My code follows this project's style guidelines
- [x] I've reviewed my own code
- [x] I've added comments for any hard-to-understand areas
- [ ] I've updated the documentation if necessary
- [x] My changes generate no new warnings
- [x] I've added tests that prove my fix is effective or my feature works
- [x] All unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published

### CODE REVIEW

File: packages/account/tests/utils_test.spec.ts [hunk 1]
```diff
@@ -0,0 +1,107 @@
+import { BiconomySmartAccountV2Config, createECDSAOwnershipValidationModule } from "../src";
```
Comment (joepegler) [inline packages/account/tests/utils_test.spec.ts:1]: could just be called utils.spect.ts

File: packages/account/src/utils/Utils.ts [hunk 2]
```diff
@@ -43,3 +47,33 @@ export function packUserOp(op: Partial<UserOperationStruct>, forSignature = true
 export const isNullOrUndefined = (value: any): value is undefined => {
   return value === null || value === undefined;
 };
+
+export const compareChainIds = async (biconomySmartAccountConfig: BiconomySmartAccountV2Config, skipChainIdCalls: boolean): Promise<Error | void> => {
```
Comment (joepegler) [inline packages/account/src/utils/Utils.ts:51]: Nice

**Review by joepegler — COMMENTED** [review]
Nice job. I think this will help clear up a lot of confusion and prevent a lot of users unnecessarily reaching out to us. Couple of small comments

**Review by joepegler — APPROVED** [review]

**Comment by livingrockrises** [comment]
I didn't get rationale behind this change? wasn't this devx abstraction improvement?

**Comment by livingrockrises** [comment]
also conflicts..

**Review by livingrockrises — COMMENTED** [review]
reviewed asked questions

**Comment by joepegler** [comment]
> I didn't get rationale behind this change? wasn't this devx abstraction improvement?

It didn't turn out to be, we've noticed a bunch of users mistakenly assuming their paymaster api key works across different chains. It's what I would have expected too. Given that the chain comes baked into the url, this change makes it plain to devs that paymaster urls don't work cross-chain, and in this PR we now throw an error when we detect a dev attempting this.

**Review by livingrockrises — APPROVED** [review]

**Pull request merged**